### PR TITLE
Fixes #14162: every agent check for postgresql and ldap partition size, resulting in possible overzealous shutting down of rudder on node

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/14162

### CODE REVIEW

File: share/commands/agent-check [hunk 1]
```diff
@@ -224,17 +227,26 @@ check_and_fix_inputs() {
...
     fi
   fi
-  if [ -d /var/lib/postgresql/ ]; then
-    psql_space=$(df /var/lib/postgresql/ | tail -n 1 | awk '{print $5}' | sed 's/%//')
-    if [ "${psql_space}" -gt 98 ]; then
-      stop_now=1
+
+  if [ -f ${RUDDER_SERVER_ROLES}/rudder-reports ]; then
+    # Get the run path of postgresql
+    PG_TABLES_PATH=$(su - postgres -c 'psql -t -c  "show data_directory;"')
```
Comment (amousset) [inline share/commands/agent-check]: smart!

**Comment by amousset** [comment]
Service to restart since 4.3 is `rudder-server`

**Comment by amousset** [comment]
We should also stop and disable rudder-agent to avoid restarting the service automatically...

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

File: share/commands/agent-check [hunk 2]
```diff
@@ -224,24 +227,76 @@ check_and_fix_inputs() {
...
+    if [ -z "${PG_TABLES_PATH}" ]; then
+      echo "FATAL: Cannot connect to PostgreSQL - please check its status" | logger -s
+    else
+      psql_space=$(df ${PG_TABLES_PATH} | tail -n 1 | awk '{print $5}' | sed 's/%//')
+
+      # If we are using TCP, we cannot stop postgresql otherwise all infra can go down - but we can stop rudder-jetty and disable rudder agent
```
Comment (peckpeck) [inline share/commands/agent-check]: In this case, postgresql will break AND the infra will go down.
hopefully we won't have this tcp problem in 5.1

**Pull request merged**